react/PageEditor.tsx

```diff
@@ -10,16 +10,18 @@ interface Props extends RenderContextProps {
   params: {
     targetPath: string
   }
-  query: Record<string, string>[]
+  query?: Record<string, string>[]
 }
 
 const PageEditor: React.FC<Props> = props => {
   const { page, params, query } = props
 
-  const queryString = Object.entries(query).reduce(
-    (acc, [key, value]) => `${acc ? acc : '?'}${`${acc}&${key}=${value}`}`,
-    ''
-  )
+  const queryString =
+    query !== undefined &&
+    Object.entries(query).reduce(
+      (acc, [key, value]) => `${acc ?? '?'}${`${acc}&${key}=${value}`}`,
+      ''
+    )
```

Comment:
```suggestion
const queryString =
  query == null
    ? ''
    : `?${Object.entries(query)
        .map(([key, value]) => `${key}=${value}`)
        .join('&')}`
```

Reply:
`==`?

Reply:
`== null` checks for `undefined` and `null` 😁

Reply:
Douglas Crockford call `==` the evil twins, here is an excerpt:

> JavaScript has two sets of equality operators: === and !==, and their evil twins == and !=. The good ones work the way you would expect. If the two operands are of the same type and have the same value, then === produces true and !== produces false. The evil twins do the right thing when the operands are of the same type, but if they are of different types, they attempt to coerce the values. the rules by which they do that are complicated and unmemorable. These are some of the interesting cases:
```js
'' == '0'           // false
0 == ''             // true
0 == '0'            // true

false == 'false'    // false
false == '0'        // true

false == undefined  // false
false == null       // false
null == undefined   // true

' \t\r\n ' == 0     // true
```

Reply:
I know, that's why we only allow the `nullish` check 😉  We use the `eqeqeq` eslint rule to prevent accidental `==` https://eslint.org/docs/rules/eqeqeq

Anyway, feel free to change that conditional, the main part was replacing the `.reduce` with a `.map.join` 😬